### PR TITLE
Indicate minimum version 2.1 for memchr dependency

### PR DESCRIPTION
**Pull request by dburgener**
The 2.0 family of memchr doesn't build in this crate.  This is hard to see using cargo minimal-versions because tokio depends on memchr 2.2, which raises the dependency using minimal-versions even when tokio is disabled.  This is a known issue with cargo-minimal-versions:

https://github.com/taiki-e/cargo-minimal-versions/issues/6

So everything looks fine from the perspective of this crate, but consumers who use quick_xml without the optional features that enable the tokio dependency have a minimal version down to 2.0, which breaks.

To reproduce in this crate, I commented out the tokio and async_tokio lines in Cargo.toml and ran cargo minimal-versions, which produces the errors, and emulates the conditions an external user might experience.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -18,7 +18,7 @@ document-features = { version = "0.2", optional = true }
 encoding_rs = { version = "0.8", optional = true }
 serde = { version = "1.0.100", optional = true }
 tokio = { version = "1.0", optional = true, default-features = false, features = ["io-util"] }
-memchr = "2.0"
+memchr = "^2.1"
```
Comment (dralley) [inline Cargo.toml:21]: The caret `^` isn't necessary, according to the reference it's equivalent to the default behavior.  Given this I would prefer it was kept consistent.

https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#caret-requirements

**Review by dralley — APPROVED** [review]

**Review by Mingun — COMMENTED** [review]
Could you provide instructions and the output of failing message? Because for me `cargo minimal-versions check` is working with `memchr = "2.0"` even when I comment `tokio`, `tokio-derive` dependencies and `asinc-tokio` feature.

**Comment by dburgener** [comment]
Interesting that you're not reproducing.  I suppose it's possible it's something weird about my environment.  Here's my exact steps, with output:

```
$ git diff
diff --git a/Cargo.toml b/Cargo.toml
index 872da6b..285a980 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -17,7 +17,7 @@ include = ["src/*", "LICENSE-MIT.md", "README.md"]
 document-features = { version = "0.2", optional = true }
 encoding_rs = { version = "0.8", optional = true }
 serde = { version = "1.0.100", optional = true }
-tokio = { version = "1.0", optional = true, default-features = false, features = ["io-util"] }
+#tokio = { version = "1.0", optional = true, default-features = false, features = ["io-util"] }
 memchr = "2.0"

 [dev-dependencies]
@@ -49,7 +49,7 @@ default = []
 ## [reading events] from types implementing [`tokio::io::AsyncBufRead`].
 ##
 ## [reading events]: crate::reader::Reader::read_event_into_async
-async-tokio = ["tokio"]
+#async-tokio = ["tokio"]

 ## Enables support of non-UTF-8 encoded documents. Encoding will be inferred from
 ## the XML declaration if it will be found, otherwise UTF-8 is assumed.
```

```
$ cargo minimal-versions check
info: running `rustup run nightly cargo update -Z minimal-versions`
    Updating crates.io index
    Removing aho-corasick v0.7.20
    Removing anes v0.1.6
    Removing async-stream v0.3.4
    Removing async-stream-impl v0.3.4
    Removing atty v0.2.14
    Removing autocfg v1.1.0
    Removing bitflags v1.3.2
    Removing bumpalo v3.12.0
    Removing bytes v1.4.0
    Removing cast v0.3.0
    Updating cfg-if v1.0.0 -> v0.1.0
    Removing ciborium v0.2.0
    Removing ciborium-io v0.2.0
    Removing ciborium-ll v0.2.0
    Removing clap v3.2.23
    Removing clap_lex v0.2.4
    Removing criterion v0.4.0
    Removing criterion-plot v0.5.0
    Removing crossbeam-channel v0.5.6
    Removing crossbeam-deque v0.8.2
    Removing crossbeam-epoch v0.9.13
    Removing crossbeam-utils v0.8.14
    Removing ctor v0.1.26
    Removing diff v0.1.13
    Updating document-features v0.2.7 -> v0.2.0
    Removing either v1.8.1
    Updating encoding_rs v0.8.32 -> v0.8.0
    Removing futures-core v0.3.26
    Removing half v1.8.2
    Removing hashbrown v0.12.3
    Removing hermit-abi v0.1.19
    Removing hermit-abi v0.2.6
    Removing indexmap v1.9.2
    Removing itertools v0.10.5
    Removing itoa v1.0.5
    Removing js-sys v0.3.61
    Removing lazy_static v1.4.0
    Updating libc v0.2.139 -> v0.2.18
    Removing litrs v0.2.3
    Removing log v0.4.17
    Updating memchr v2.5.0 -> v2.0.0
    Removing memoffset v0.7.1
    Removing num-traits v0.2.15
    Removing num_cpus v1.15.0
    Removing once_cell v1.17.1
    Removing oorandom v11.1.3
    Removing ordered-float v2.10.0
    Removing os_str_bytes v6.4.1
    Removing output_vt100 v0.1.3
    Removing pin-project-lite v0.2.9
    Removing plotters v0.3.4
    Removing plotters-backend v0.3.4
    Removing plotters-svg v0.3.3
    Removing pretty_assertions v1.3.0
    Removing proc-macro2 v1.0.51
    Updating quote v1.0.23 -> v0.3.8
    Removing rayon v1.6.1
    Removing rayon-core v1.10.2
    Removing regex v1.7.1
    Removing regex-syntax v0.6.28
    Removing ryu v1.0.12
    Removing same-file v1.0.6
    Removing scopeguard v1.1.0
    Updating serde v1.0.152 -> v1.0.100
    Removing serde-value v0.7.0
    Updating serde_derive v1.0.152 -> v1.0.0
      Adding serde_derive_internals v0.15.0
    Removing serde_json v1.0.93
    Updating syn v1.0.108 -> v0.11.10
      Adding synom v0.11.0
    Removing textwrap v0.16.0
    Removing tinytemplate v1.2.1
    Removing tokio v1.25.0
    Removing tokio-macros v1.8.2
    Removing tokio-stream v0.1.12
    Removing tokio-test v0.4.2
    Removing unicode-ident v1.0.6
      Adding unicode-xid v0.0.4
    Removing walkdir v2.3.2
    Removing wasm-bindgen v0.2.84
    Removing wasm-bindgen-backend v0.2.84
    Removing wasm-bindgen-macro v0.2.84
    Removing wasm-bindgen-macro-support v0.2.84
    Removing wasm-bindgen-shared v0.2.84
    Removing web-sys v0.3.61
    Removing winapi v0.3.9
    Removing winapi-i686-pc-windows-gnu v0.4.0
    Removing winapi-util v0.1.5
    Removing winapi-x86_64-pc-windows-gnu v0.4.0
    Removing windows-sys v0.42.0
    Removing windows_aarch64_gnullvm v0.42.1
    Removing windows_aarch64_msvc v0.42.1
    Removing windows_i686_gnu v0.42.1
    Removing windows_i686_msvc v0.42.1
    Removing windows_x86_64_gnu v0.42.1
    Removing windows_x86_64_gnullvm v0.42.1
    Removing windows_x86_64_msvc v0.42.1
    Removing yansi v0.5.1
info: running `cargo hack check`
info: running `cargo check` on quick-xml (1/1)
    Checking quick-xml v0.27.1 (/home/dburgener/quick-xml)
error[E0432]: unresolved import `memchr::memchr2_iter`
 --> src/escapei.rs:3:5
  |
3 | use memchr::memchr2_iter;
  |     ^^^^^^^^^^^^^^^^^^^^ no `memchr2_iter` in the root

error[E0425]: cannot find function `memchr_iter` in crate `memchr`
  --> src/reader/parser.rs:96:46
   |
96 |                     if let Some(p) = memchr::memchr_iter(b'-', &buf[3..len - 2])
   |                                              ^^^^^^^^^^^ not found in `memchr`

error[E0425]: cannot find function `memchr_iter` in crate `memchr`
   --> src/reader/mod.rs:773:26
    |
773 |         for i in memchr::memchr_iter(b'>', chunk) {
    |                          ^^^^^^^^^^^ not found in `memchr`

error[E0425]: cannot find function `memchr2_iter` in crate `memchr`
   --> src/reader/mod.rs:815:43
    |
815 |                     let balance = memchr::memchr2_iter(b'<', b'>', content)
    |                                           ^^^^^^^^^^^^ not found in `memchr`

error[E0425]: cannot find function `memchr3_iter` in crate `memchr`
   --> src/reader/mod.rs:853:26
    |
853 |         for i in memchr::memchr3_iter(b'>', b'\'', b'"', chunk) {
    |                          ^^^^^^^^^^^^ not found in `memchr`

Some errors have detailed explanations: E0425, E0432.
For more information about an error, try `rustc --explain E0425`.
error: could not compile `quick-xml` due to 5 previous errors
error: process didn't exit successfully: `/home/dburgener/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/bin/cargo check --manifest-path Cargo.toml` (exit status: 101)
error: process didn't exit successfully: `/home/dburgener/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/bin/cargo hack check` (exit status: 1)
```

**Comment by dburgener** [comment]
I also tried with the dev dependencies commented out as it seems that you did, but I still get the same memchr compile errors:

```
$ git diff
diff --git a/Cargo.toml b/Cargo.toml
index 872da6b..7611b27 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -17,7 +17,7 @@ include = ["src/*", "LICENSE-MIT.md", "README.md"]
 document-features = { version = "0.2", optional = true }
 encoding_rs = { version = "0.8", optional = true }
 serde = { version = "1.0.100", optional = true }
-tokio = { version = "1.0", optional = true, default-features = false, features = ["io-util"] }
+#tokio = { version = "1.0", optional = true, default-features = false, features = ["io-util"] }
 memchr = "2.0"

 [dev-dependencies]
@@ -26,8 +26,8 @@ pretty_assertions = "1.3"
 regex = "1"
 serde = { version = "1.0", features = ["derive"] }
 serde-value = "0.7"
-tokio = { version = "1.21", default-features = false, features = ["macros", "rt"] }
-tokio-test = "0.4"
+#tokio = { version = "1.21", default-features = false, features = ["macros", "rt"] }
+#tokio-test = "0.4"

 [lib]
 bench = false
@@ -49,7 +49,7 @@ default = []
 ## [reading events] from types implementing [`tokio::io::AsyncBufRead`].
 ##
 ## [reading events]: crate::reader::Reader::read_event_into_async
-async-tokio = ["tokio"]
+#async-tokio = ["tokio"]

 ## Enables support of non-UTF-8 encoded documents. Encoding will be inferred from
 ## the XML declaration if it will be found, otherwise UTF-8 is assumed.
```

```
$ cargo minimal-versions check
info: running `rustup run nightly cargo update -Z minimal-versions`
    Updating crates.io index
    Removing aho-corasick v0.7.20
    Removing anes v0.1.6
    Removing async-stream v0.3.4
    Removing async-stream-impl v0.3.4
    Removing atty v0.2.14
    Removing autocfg v1.1.0
    Removing bitflags v1.3.2
    Removing bumpalo v3.12.0
    Removing bytes v1.4.0
    Removing cast v0.3.0
    Updating cfg-if v1.0.0 -> v0.1.0
    Removing ciborium v0.2.0
    Removing ciborium-io v0.2.0
    Removing ciborium-ll v0.2.0
    Removing clap v3.2.23
    Removing clap_lex v0.2.4
    Removing criterion v0.4.0
    Removing criterion-plot v0.5.0
    Removing crossbeam-channel v0.5.6
    Removing crossbeam-deque v0.8.2
    Removing crossbeam-epoch v0.9.13
    Removing crossbeam-utils v0.8.14
    Removing ctor v0.1.26
    Removing diff v0.1.13
    Updating document-features v0.2.7 -> v0.2.0
    Removing either v1.8.1
    Updating encoding_rs v0.8.32 -> v0.8.0
    Removing futures-core v0.3.26
    Removing half v1.8.2
    Removing hashbrown v0.12.3
    Removing hermit-abi v0.1.19
    Removing hermit-abi v0.2.6
    Removing indexmap v1.9.2
    Removing itertools v0.10.5
    Removing itoa v1.0.5
    Removing js-sys v0.3.61
    Removing lazy_static v1.4.0
    Updating libc v0.2.139 -> v0.2.18
    Removing litrs v0.2.3
    Removing log v0.4.17
    Updating memchr v2.5.0 -> v2.0.0
    Removing memoffset v0.7.1
    Removing num-traits v0.2.15
    Removing num_cpus v1.15.0
    Removing once_cell v1.17.1
    Removing oorandom v11.1.3
    Removing ordered-float v2.10.0
    Removing os_str_bytes v6.4.1
    Removing output_vt100 v0.1.3
    Removing pin-project-lite v0.2.9
    Removing plotters v0.3.4
    Removing plotters-backend v0.3.4
    Removing plotters-svg v0.3.3
    Removing pretty_assertions v1.3.0
    Removing proc-macro2 v1.0.51
    Updating quote v1.0.23 -> v0.3.8
    Removing rayon v1.6.1
    Removing rayon-core v1.10.2
    Removing regex v1.7.1
    Removing regex-syntax v0.6.28
    Removing ryu v1.0.12
    Removing same-file v1.0.6
    Removing scopeguard v1.1.0
    Updating serde v1.0.152 -> v1.0.100
    Removing serde-value v0.7.0
    Updating serde_derive v1.0.152 -> v1.0.0
      Adding serde_derive_internals v0.15.0
    Removing serde_json v1.0.93
    Updating syn v1.0.108 -> v0.11.10
      Adding synom v0.11.0
    Removing textwrap v0.16.0
    Removing tinytemplate v1.2.1
    Removing tokio v1.25.0
    Removing tokio-macros v1.8.2
    Removing tokio-stream v0.1.12
    Removing tokio-test v0.4.2
    Removing unicode-ident v1.0.6
      Adding unicode-xid v0.0.4
    Removing walkdir v2.3.2
    Removing wasm-bindgen v0.2.84
    Removing wasm-bindgen-backend v0.2.84
    Removing wasm-bindgen-macro v0.2.84
    Removing wasm-bindgen-macro-support v0.2.84
    Removing wasm-bindgen-shared v0.2.84
    Removing web-sys v0.3.61
    Removing winapi v0.3.9
    Removing winapi-i686-pc-windows-gnu v0.4.0
    Removing winapi-util v0.1.5
    Removing winapi-x86_64-pc-windows-gnu v0.4.0
    Removing windows-sys v0.42.0
    Removing windows_aarch64_gnullvm v0.42.1
    Removing windows_aarch64_msvc v0.42.1
    Removing windows_i686_gnu v0.42.1
    Removing windows_i686_msvc v0.42.1
    Removing windows_x86_64_gnu v0.42.1
    Removing windows_x86_64_gnullvm v0.42.1
    Removing windows_x86_64_msvc v0.42.1
    Removing yansi v0.5.1
info: running `cargo hack check`
info: running `cargo check` on quick-xml (1/1)
    Checking quick-xml v0.27.1 (/home/dburgener/quick-xml)
error[E0432]: unresolved import `memchr::memchr2_iter`
 --> src/escapei.rs:3:5
  |
3 | use memchr::memchr2_iter;
  |     ^^^^^^^^^^^^^^^^^^^^ no `memchr2_iter` in the root

error[E0425]: cannot find function `memchr_iter` in crate `memchr`
  --> src/reader/parser.rs:96:46
   |
96 |                     if let Some(p) = memchr::memchr_iter(b'-', &buf[3..len - 2])
   |                                              ^^^^^^^^^^^ not found in `memchr`

error[E0425]: cannot find function `memchr_iter` in crate `memchr`
   --> src/reader/mod.rs:773:26
    |
773 |         for i in memchr::memchr_iter(b'>', chunk) {
    |                          ^^^^^^^^^^^ not found in `memchr`

error[E0425]: cannot find function `memchr2_iter` in crate `memchr`
   --> src/reader/mod.rs:815:43
    |
815 |                     let balance = memchr::memchr2_iter(b'<', b'>', content)
    |                                           ^^^^^^^^^^^^ not found in `memchr`

error[E0425]: cannot find function `memchr3_iter` in crate `memchr`
   --> src/reader/mod.rs:853:26
    |
853 |         for i in memchr::memchr3_iter(b'>', b'\'', b'"', chunk) {
    |                          ^^^^^^^^^^^^ not found in `memchr`

Some errors have detailed explanations: E0425, E0432.
For more information about an error, try `rustc --explain E0425`.
error: could not compile `quick-xml` due to 5 previous errors
error: process didn't exit successfully: `/home/dburgener/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/bin/cargo check --manifest-path Cargo.toml` (exit status: 101)
error: process didn't exit successfully: `/home/dburgener/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/bin/cargo hack check` (exit status: 1)
```

**Comment by dburgener** [comment]
As far as I can tell, memchr*_iter() were all added in this memchr commit: https://github.com/BurntSushi/memchr/commit/3120980c1f168cccb2bfc0fbaa7e9c5288701e05, which first was in memchr 2.1, so although it very confusing to me how it is not reproducing on your end, I do think that 2.1 seems like the correct minimum version for this dependency.

**Comment by Mingun** [comment]
Hm, it seems for me cargo somehow taken the lib from the cache. Could you add a change log entry and then we can merge.

**Comment by dburgener** [comment]
> Could you add a change log entry and then we can merge.

Yes, done.

File: Changelog.md [hunk 2]
```diff
@@ -32,6 +32,7 @@
   `DeEvent::CData` events was replaced by `DeEvent::Text` with merged content.
   The same behavior for the `Reader` does not implemented (yet?) and should be
   implemented manually
+- [#562]: Correctly set minimum required version of memchr dependency to 2.1
```
Comment (Mingun) [inline Changelog.md:35]: You forgot to add a link target at line 47 (keep list of targets sorted, please)

Comment (dburgener) [reply Changelog.md:35]: fixed, thanks.

**Comment by Mingun** [comment]
Ok, at least I understand how I get the false positive result. `cargo minimal-versions check` does not see any problems, but `cargo minimal-versions test` found the problems. Very strange behavior.

**Comment by dburgener** [comment]
Very strange.  I didn't try `cargo minimal-versions test`.  It does seem to me as though there are some weird corner cases around optional dependencies in minimal-versions that aren't fully sorted out, so this is likely an aspect of that.  In theory, the commenting around Cargo.toml shouldn't be necessary at all, and just running `cargo minimal-versions check --no-default-features` (or `--each-feature`) should catch this case consistently and cleanly, but that isn't the case.

What's strange to me is that based on the output [here](https://github.com/taiki-e/cargo-hack/issues/183) it looks like you were getting down to memchr 2.0, which shouldn't have compiled:

```
Updating memchr v2.5.0 -> v2.0.0
```

Unless that changed once you got the issue you were mentioning at that link sorted out?

**Comment by Mingun** [comment]
Maybe this is the nightly regression in cargo? I use nightly `cargo` with a GNU toolchain and due to inability to install MSVC tools cannot check on stable, because `cargo minimum-versions` tried to switch to `nightly` which on Windows assumes `nightly-x86_64-pc-windows-msvc` which I cannot install...

I filled a bug in cargo for that.

**Comment by dburgener** [comment]
I think I can repro on nightly as well:

```
$ git diff
diff --git a/Cargo.toml b/Cargo.toml
index 872da6b..7611b27 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -17,7 +17,7 @@ include = ["src/*", "LICENSE-MIT.md", "README.md"]
 document-features = { version = "0.2", optional = true }
 encoding_rs = { version = "0.8", optional = true }
 serde = { version = "1.0.100", optional = true }
-tokio = { version = "1.0", optional = true, default-features = false, features = ["io-util"] }
+#tokio = { version = "1.0", optional = true, default-features = false, features = ["io-util"] }
 memchr = "2.0"

 [dev-dependencies]
@@ -26,8 +26,8 @@ pretty_assertions = "1.3"
 regex = "1"
 serde = { version = "1.0", features = ["derive"] }
 serde-value = "0.7"
-tokio = { version = "1.21", default-features = false, features = ["macros", "rt"] }
-tokio-test = "0.4"
+#tokio = { version = "1.21", default-features = false, features = ["macros", "rt"] }
+#tokio-test = "0.4"

 [lib]
 bench = false
@@ -49,7 +49,7 @@ default = []
 ## [reading events] from types implementing [`tokio::io::AsyncBufRead`].
 ##
 ## [reading events]: crate::reader::Reader::read_event_into_async
-async-tokio = ["tokio"]
+#async-tokio = ["tokio"]

 ## Enables support of non-UTF-8 encoded documents. Encoding will be inferred from
 ## the XML declaration if it will be found, otherwise UTF-8 is assumed.
```

```
$ rustup run nightly cargo minimal-versions check
info: running `cargo update -Z minimal-versions`
    Updating crates.io index
    Removing aho-corasick v0.7.20
    Removing anes v0.1.6
    Removing async-stream v0.3.4
    Removing async-stream-impl v0.3.4
    Removing atty v0.2.14
    Removing autocfg v1.1.0
    Removing bitflags v1.3.2
    Removing bumpalo v3.12.0
    Removing bytes v1.4.0
    Removing cast v0.3.0
    Updating cfg-if v1.0.0 -> v0.1.0
    Removing ciborium v0.2.0
    Removing ciborium-io v0.2.0
    Removing ciborium-ll v0.2.0
    Removing clap v3.2.23
    Removing clap_lex v0.2.4
    Removing criterion v0.4.0
    Removing criterion-plot v0.5.0
    Removing crossbeam-channel v0.5.6
    Removing crossbeam-deque v0.8.2
    Removing crossbeam-epoch v0.9.13
    Removing crossbeam-utils v0.8.14
    Removing ctor v0.1.26
    Removing diff v0.1.13
    Updating document-features v0.2.7 -> v0.2.0
    Removing either v1.8.1
    Updating encoding_rs v0.8.32 -> v0.8.0
    Removing futures-core v0.3.26
    Removing half v1.8.2
    Removing hashbrown v0.12.3
    Removing hermit-abi v0.1.19
    Removing hermit-abi v0.2.6
    Removing indexmap v1.9.2
    Removing itertools v0.10.5
    Removing itoa v1.0.5
    Removing js-sys v0.3.61
    Removing lazy_static v1.4.0
    Updating libc v0.2.139 -> v0.2.18
    Removing litrs v0.2.3
    Removing log v0.4.17
    Updating memchr v2.5.0 -> v2.0.0
    Removing memoffset v0.7.1
    Removing num-traits v0.2.15
    Removing num_cpus v1.15.0
    Removing once_cell v1.17.1
    Removing oorandom v11.1.3
    Removing ordered-float v2.10.0
    Removing os_str_bytes v6.4.1
    Removing output_vt100 v0.1.3
    Removing pin-project-lite v0.2.9
    Removing plotters v0.3.4
    Removing plotters-backend v0.3.4
    Removing plotters-svg v0.3.3
    Removing pretty_assertions v1.3.0
    Removing proc-macro2 v1.0.51
    Updating quote v1.0.23 -> v0.3.8
    Removing rayon v1.6.1
    Removing rayon-core v1.10.2
    Removing regex v1.7.1
    Removing regex-syntax v0.6.28
    Removing ryu v1.0.12
    Removing same-file v1.0.6
    Removing scopeguard v1.1.0
    Updating serde v1.0.152 -> v1.0.100
    Removing serde-value v0.7.0
    Updating serde_derive v1.0.152 -> v1.0.0
      Adding serde_derive_internals v0.15.0
    Removing serde_json v1.0.93
    Updating syn v1.0.108 -> v0.11.10
      Adding synom v0.11.0
    Removing textwrap v0.16.0
    Removing tinytemplate v1.2.1
    Removing tokio v1.25.0
    Removing tokio-macros v1.8.2
    Removing tokio-stream v0.1.12
    Removing tokio-test v0.4.2
    Removing unicode-ident v1.0.6
      Adding unicode-xid v0.0.4
    Removing walkdir v2.3.2
    Removing wasm-bindgen v0.2.84
    Removing wasm-bindgen-backend v0.2.84
    Removing wasm-bindgen-macro v0.2.84
    Removing wasm-bindgen-macro-support v0.2.84
    Removing wasm-bindgen-shared v0.2.84
    Removing web-sys v0.3.61
    Removing winapi v0.3.9
    Removing winapi-i686-pc-windows-gnu v0.4.0
    Removing winapi-util v0.1.5
    Removing winapi-x86_64-pc-windows-gnu v0.4.0
    Removing windows-sys v0.42.0
    Removing windows_aarch64_gnullvm v0.42.1
    Removing windows_aarch64_msvc v0.42.1
    Removing windows_i686_gnu v0.42.1
    Removing windows_i686_msvc v0.42.1
    Removing windows_x86_64_gnu v0.42.1
    Removing windows_x86_64_gnullvm v0.42.1
    Removing windows_x86_64_msvc v0.42.1
    Removing yansi v0.5.1
info: running `cargo hack check`
info: running `cargo check` on quick-xml (1/1)
    Checking libc v0.2.18
    Checking memchr v2.0.0
    Checking quick-xml v0.27.1 (/home/dburgener/quick-xml)
error[E0432]: unresolved import `memchr::memchr2_iter`
 --> src/escapei.rs:3:5
  |
3 | use memchr::memchr2_iter;
  |     ^^^^^^^^^^^^^^^^^^^^ no `memchr2_iter` in the root

error[E0425]: cannot find function `memchr_iter` in crate `memchr`
  --> src/reader/parser.rs:96:46
   |
96 |                     if let Some(p) = memchr::memchr_iter(b'-', &buf[3..len - 2])
   |                                              ^^^^^^^^^^^ not found in `memchr`

error[E0425]: cannot find function `memchr_iter` in crate `memchr`
   --> src/reader/mod.rs:773:26
    |
773 |         for i in memchr::memchr_iter(b'>', chunk) {
    |                          ^^^^^^^^^^^ not found in `memchr`

error[E0425]: cannot find function `memchr2_iter` in crate `memchr`
   --> src/reader/mod.rs:815:43
    |
815 |                     let balance = memchr::memchr2_iter(b'<', b'>', content)
    |                                           ^^^^^^^^^^^^ not found in `memchr`

error[E0425]: cannot find function `memchr3_iter` in crate `memchr`
   --> src/reader/mod.rs:853:26
    |
853 |         for i in memchr::memchr3_iter(b'>', b'\'', b'"', chunk) {
    |                          ^^^^^^^^^^^^ not found in `memchr`

Some errors have detailed explanations: E0425, E0432.
For more information about an error, try `rustc --explain E0425`.
error: could not compile `quick-xml` due to 5 previous errors
error: process didn't exit successfully: `/home/dburgener/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/bin/cargo check --manifest-path Cargo.toml` (exit status: 101)
error: process didn't exit successfully: `/home/dburgener/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/bin/cargo hack check` (exit status: 1)
```

**Comment by Mingun** [comment]
It seems that VSCode with rust-analyzer is guilty (RA updates `Cargo.lock` to latest versions after downgrading versions but before actual check).
